sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -81,7 +82,16 @@ public InformationSchemaIterables(final Schemas schemas,
         partitionInfos = new PartitionInfos(clusterService);
         columnsIterable = tablesIterable.transformAndConcat(ColumnsIterable::new);
 
-        constraints = tablesIterable.filter(i -> i != null && i.primaryKey().size() > 0);
+        // Check if primary key is defined and is not _id.
+        FluentIterable<ConstraintInfo> primaryKeyConstraints = tablesIterable
+            .filter(i -> i != null && (i.primaryKey().size() > 1 ||
+                        (i.primaryKey().size() == 1 && i.primaryKey().get(0).name() != "_id")))
```

Comment:
This is comparing Strings by reference which only works because of JVM String pooling. `equals(..)` is safer here.